src/pages/RegPage.jsx

```diff
@@ -0,0 +1,101 @@
+import React, { useReducer, useState } from "react";
+import DisplayRequest from "../views/DisplayRequest";
+import FlowData from "../views/FlowData";
+import API, { getFormattedURL, getFormattedProxyURL } from "../API";
+import RegistrationProvider from "../util/RegistrationProvider";
+import { Container, Row, Col } from "react-bootstrap";
+import Registration from "../views/Registration";
+
+const RegPage = (props) => {
+    const reset = () => {
+        setRequest({});
+        setResponse({});
+        setCSRF("");
+        setFlowID("");
+        setSessionID("");
+    }
+
+    const resetPrint = () => {
+        setRequest({});
+        setResponse({});
+    }
+
+    const [request, setRequest] = useState({});
+
+    const [response, setResponse] = useState({});
+
+    const [flowID, setFlowID] = useState("");
+    const [csrf, setCSRF] = useState("");
+    const [accountKey, setAccountKey] = useState("");
+    const [sessionID, setSessionID] = useState("");
+
+    const initRegistration = async () => {
+        reset();
+
+        let registration = new RegistrationProvider(props.cluster, props.flow);
+        let newRequest = registration.getInitRequest();
+
+        setRequest(newRequest);
+
+        var jsonRes = await API.makeRequest(newRequest);
+        setResponse({ ...response, headers: jsonRes.headers, status: jsonRes.status, body: jsonRes.json });
+        setFlowID(jsonRes.json.id);
+        setCSRF(jsonRes.json.ui.nodes.filter(x => x.attributes.name == "csrf_token").map(x => x.attributes.value)[0]);
```

Comment:
```suggestion
        const request = registration.getInitRequest();
        updateState({ request: request });

        const jsonRes = await API.makeRequest(request);
        const csrf = jsonRes.json.ui.nodes.filter(x => x.attributes.name == "csrf_token").map(x => x.attributes.value)[0];
        updateState({
            flowID: jsonRes.json.id,
            csrf: csrf,
            request: request,
            response: { headers: jsonRes.headers, status: jsonRes.status, body: jsonRes.json },
        });
```

Update to the new state management. I'll check if there is an easier way to get the `csrf` token from the response :D